quilter.py

```diff
@@ -0,0 +1,360 @@
+parser.add_argument('destination', 
+                    help='Directory for extracted files (must already exist).')
+
+parser.add_argument('-m', '--merge', dest="name", 
+                    help='Merge downloaded files to specified base name. .tif\
+                    or .shp extensions will be added as appropriate. Requires\
+                    GDAL.')
+parser.add_argument('-p', '--project', dest="crs", help='Reproject merged file\
+                    to specified CRS. Specify CRS like EPSG:x or ESRI:x.\
+                    Requires -m.')
+parser.add_argument('-d', '--dont_compress', help='Disable LZW compression of\
+                    merged raster (all aerials will remain compressed).',
+                    action='store_true')
+
+args = parser.parse_args()
+
+# Set defaults
+merge = False
+project = False
+compress = True
+delete = False
+
+outfolder = args.destination
+csv_file = args.csv
+if args.name:
+    final_name = args.name
+    merge = True
+if args.crs:
+    projection = args.crs
+    project = True
+if args.dont_compress:
+    compress = False
+
+# outfolder = 'c:\\gis\\elevation\\statewide\\test'
+# csv_file = 'c:\\users\\jdadams\\downloads\\10mDEMS2.csv'
+# projection = "EPSG:26912"
+# final_name = "10m"
+
+ext_list = ('.img', '.tif', '.asc', '.jpg', '.jgw', '.xyz', '.tfw', '.dbf', 
+            '.prj', '.shp', '.shx')
+raster_exts = ('.img', '.tif', '.asc', '.jpg')
+
+# merge = True
+# project = True 
+# compress = True  # Default to always compress rasters w/ lzw
+
+# TODO: Need to set these automagically. Maybe something in CSV?
+raster = True
+vector = False
+
+
+# Progress bar for download
+# modified from https://sumit-ghosh.com/articles/python-download-progress-bar/
+def progbar(progress, total):
+    done = int(50 * progress / total)
+    percent = round(100 * progress / total, 2)
+    sys.stdout.write('\r[{}{}] {}%'.format('#' * done, '_' * (50 - done),
+                                           percent))
+    sys.stdout.flush()
+
+
+# Shamelessly modified from Sumit Ghosh, 
+# https://sumit-ghosh.com/articles/python-download-progress-bar/
+def download(url, filename):
+    with open(filename, 'wb') as f:
+        response = requests.get(url, stream=True)
+        total = response.headers.get('content-length')
+
+        if total is None:
+            f.write(response.content)
+        else:
+            downloaded = 0
+            total = int(total)
+            for data in response.iter_content(chunk_size=max(int(total/1000), 1024*1024)):
+                downloaded += len(data)
+                f.write(data)
+                progbar(downloaded, total)
+    sys.stdout.write('\n')
+
+
+# GDAL callback method that seems to work, as per
+# https://gis.stackexchange.com/questions/237479/using-callback-with-python-gdal-rasterizelayer
+def gdal_progress_callback(complete, message, unknown):
+    # 40 stops on our progress bar, so scale to 40
+    done = int(40 * complete / 1)
```

Comment:
this seems like a really long method, do you think it would benefit from being broken down into smaller chunks for eventual unit tests?

Reply:
Which method are you referring to? gdal_progress_callback() is 16 lines. If you're referring to the whole script, perhaps. This is very much an in-process script, and I did a PR to get things started.